DependencyInjection/Configuration.php

```diff
@@ -41,24 +42,80 @@ public function getConfigTreeBuilder() : TreeBuilder
+                    ->useAttributeAsKey('name')
+                    ->defaultValue(['%kernel.root_dir%/DoctrineMigrations' => 'Application\Migrations'])
+                    ->prototype('scalar')->end()
+                    ->validate()
+                        ->ifTrue(static function ($v) {
+                            return count($v) === 0;
+                        })
+                        ->thenInvalid('At least one migrations path must be specified.')
+                    ->end()
+                 ->end()
+
+                ->arrayNode('storage')
+                    ->addDefaultsIfNotSet()
+                    ->info('Storage to use for migration status metadata.')
+                    ->children()
+                        ->scalarNode('id')
+                            ->info('Custom metadata storage service ID.')
+                            ->defaultValue(null)
+                        ->end()
+                        ->arrayNode('table_storage')
+                            ->addDefaultsIfNotSet()
+                            ->info('The default metadata storage, implemented as table in the database.')
+                            ->children()
+                                ->scalarNode('table_name')->defaultValue(null)->cannotBeEmpty()->end()
+                                ->scalarNode('version_column_name')->defaultValue(null)->end()
+                                ->scalarNode('version_column_length')->defaultValue(null)->end()
+                                ->scalarNode('executed_at_column_name')->defaultValue(null)->end()
+                                ->scalarNode('execution_time_column_name')->defaultValue(null)->end()
+                            ->end()
+                        ->end()
+                    ->end()
+
+                ->end()
+                ->arrayNode('migrations')
+                    ->info('A list of migrations to load in addition the the one discovered via "migrations_paths".')
+                    ->prototype('scalar')->end()
+                    ->defaultValue([])
+                ->end()
+                ->scalarNode('sorter')
+                    ->info('Alternative migrations sorting algorithm')
+                    ->defaultValue(null)
+                ->end()
+                ->scalarNode('connection')
+                    ->info('Connection name to use for the migrations database.')
+                    ->defaultValue(null)
+                ->end()
+                ->scalarNode('em')
+                    ->info('Entity manager name to use for the migrations database (available when doctrine/orm is installed).')
+                    ->defaultValue(null)
+                ->end()
+                ->scalarNode('all_or_nothing')
+                    ->info('Run all migrations in a transaction.')
+                    ->defaultValue(false)
+                ->end()
+                ->scalarNode('check_database_platform')->defaultValue(true)->end()
```

Comment:
I think this option definitely warrants an explanatory comment.

Reply:
done